genesyscloud/telephony_providers_edges_phonebasesettings/genesyscloud_telephony_providers_edges_phonebasesettings_proxy.go

```diff
@@ -0,0 +1,130 @@
+// that we can still proxy our tests by directly setting internalProxy package variable
+func getPhoneBaseProxy(clientConfig *platformclientv2.Configuration) *phoneBaseProxy {
+	if internalProxy == nil {
+		internalProxy = newphoneBaseProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+func (p *phoneBaseProxy) getPhoneBaseSetting(ctx context.Context, phoneBaseSettingsId string) (*platformclientv2.Phonebase, *platformclientv2.APIResponse, error) {
+	return p.getPhoneBaseSettingAttr(ctx, p, phoneBaseSettingsId)
+}
+
+func (p *phoneBaseProxy) deletePhoneBaseSetting(ctx context.Context, phoneBaseSettingsId string) (*platformclientv2.APIResponse, error) {
+	return p.deletePhoneBaseSettingAttr(ctx, p, phoneBaseSettingsId)
+}
+
+func (p *phoneBaseProxy) putPhoneBaseSetting(ctx context.Context, phoneBaseSettingsId string, body platformclientv2.Phonebase) (*platformclientv2.Phonebase, *platformclientv2.APIResponse, error) {
+	return p.putPhoneBaseSettingAttr(ctx, p, phoneBaseSettingsId, body)
+}
+
+func (p *phoneBaseProxy) postPhoneBaseSetting(ctx context.Context, body platformclientv2.Phonebase) (*platformclientv2.Phonebase, *platformclientv2.APIResponse, error) {
+	return p.postPhoneBaseSettingAttr(ctx, p, body)
+}
+
+func (p *phoneBaseProxy) getAllPhoneBaseSettings(ctx context.Context) (*[]platformclientv2.Phonebase, error) {
+	return p.getAllPhoneBaseSettingsAttr(ctx, p)
+}
+
+// getPhoneBaseSettingFn is an implementation function for retrieving a Genesys Cloud Phone Base Setting
+func getPhoneBaseSettingFn(ctx context.Context, p *phoneBaseProxy, phoneBaseSettingsId string) (*platformclientv2.Phonebase, *platformclientv2.APIResponse, error) {
+	phoneBase, resp, err := p.edgesApi.GetTelephonyProvidersEdgesPhonebasesetting(phoneBaseSettingsId)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return phoneBase, resp, nil
+}
+
+func deletePhoneBaseSettingsFn(ctx context.Context, p *phoneBaseProxy, phoneBaseSettingsId string) (*platformclientv2.APIResponse, error) {
+	resp, err := p.edgesApi.DeleteTelephonyProvidersEdgesPhonebasesetting(phoneBaseSettingsId)
+	return resp, err
+}
+
+func putPhoneBaseSettingFn(ctx context.Context, p *phoneBaseProxy, phoneBaseSettingsId string, body platformclientv2.Phonebase) (*platformclientv2.Phonebase, *platformclientv2.APIResponse, error) {
+	phoneBase, resp, err := p.edgesApi.PutTelephonyProvidersEdgesPhonebasesetting(phoneBaseSettingsId, body)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return phoneBase, resp, nil
+}
+
+func postPhoneBaseSettingFn(ctx context.Context, p *phoneBaseProxy, body platformclientv2.Phonebase) (*platformclientv2.Phonebase, *platformclientv2.APIResponse, error) {
+	phoneBase, resp, err := p.edgesApi.PostTelephonyProvidersEdgesPhonebasesettings(body)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return phoneBase, resp, nil
+}
+
+func getAllPhoneBaseSettingsFn(ctx context.Context, p *phoneBaseProxy) (*[]platformclientv2.Phonebase, error) {
```

Comment:
if possible add in a few unit tests for this method

Reply:
this story is for proxy only

Reply:
proxy tests would be considered part of the proxy imo

Reply:
https://inindca.atlassian.net/browse/DEVTOOLING-325